subprojects/core/src/main/java/org/gradle/api/Project.java

```diff
@@ -901,6 +903,42 @@
     FileTree tarTree(Object tarPath);
 
     /**
+     * Creates a {@code Provider} implementation based on the provided class.
+     * The value returned by the provider is represented by the default value of the standard Java data type.
+     * Please see <a href="https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html">Oracle's Java manual</a> for more information.
+     * For the Gradle file types {@code FileCollection} and {@code FileTree} the value returned by the provider represents an empty list of files.
+     *
+     * @param clazz The class to be used for provider.
+     * @return The provider. Never returns null.
+     * @throws org.gradle.api.InvalidUserDataException If the provided class is null.
+     * @since 3.5
+     */
+    @Incubating
+    <T> Provider<T> defaultProvider(Class<T> clazz);
+
+    /**
+     * Creates a new {@code Provider} for the Closure that lazily evaluates to a specific value.
+     *
+     * @param value The value to be used for type returned by the provider.
+     * @return The provider. Never returns null.
+     * @throws org.gradle.api.InvalidUserDataException If the provided value is null.
```

Comment:
I will add a test case for `null`. There was already a test case for the exception: `ProviderFactoryTest.rethrows checked exception as unchecked for lazily evaluated value`.